pdns/slavecommunicator.cc

```diff
@@ -992,7 +996,7 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
     }
     else if(hasSOA && theirserial == ourserial) {
       uint32_t maxExpire=0, maxInception=0;
-      if(dk.isPresigned(di.zone)) {
+      if(dk.isPresigned(di.zone) && ::arg().mustDo("compare-signatures-on-zone-freshness-check")) {
```

Comment:
I think it would be cleaner if the "dnssecOk flag" came out of `val` (= `Inflighter<vector<DomainNotificationInfo>, SlaveSenderReceiver>` return data), instead of rereading the arg().